cmake/KokkosConfig.cmake.in

```diff
@@ -33,3 +33,16 @@ IF(@Kokkos_ENABLE_CUDA@ AND NOT "separable_compilation" IN_LIST Kokkos_FIND_COMP
     ENDIF()
     UNSET(IS_NVCC) # be mindful of the environment, pollution is bad
 ENDIF()
+
+# It is highly recommended that projects using Kokkos use the same compiler as when Kokkos was installed but HPC commonly has
+# complex toolchains where different compiler IDs are necessary:
+# - If separable_compilation is requested, i.e. find_package(Kokkos COMPONENTS separable_compilation), the user 
+#   is assuming control of redirection to the compiler for kokkos code so no warning should ever be emitted.
+# - Emitting the warning during find_package(Kokkos QUIET) is not finding the package quietly
+# - AUTHOR_WARNING instead of WARNING enables setting CMAKE_SUPPRESS_DEVELOPER_WARNINGS to ON via the command-line or
+#   locally in a cmake script to suppress this message  
+IF(NOT "separable_compilation" IN_LIST Kokkos_FIND_COMPONENTS AND 
```

Comment:
There is a new `"launch_compiler"` component in addition to `"separable_compilation"`. I think you need to sync your branch